Sources/Public/BacktraceBreadcrumbsLogManager.swift

```diff
@@ -0,0 +1,52 @@
+import Foundation
+
+@objc public class BacktraceBreadcrumbsLogManager: NSObject {
+    
+    private var breadcrumbId = Date().millisecondsSince1970
+    
+    private var backtraceQueueFileHelper: BacktraceQueueFileHelper?
+
+    public init(_ breadcrumbLogPath: String, maxQueueFileSizeBytes: Int) {
+        super.init()
+        self.backtraceQueueFileHelper = BacktraceQueueFileHelper(breadcrumbLogPath, maxQueueFileSizeBytes: maxQueueFileSizeBytes)
+    }
+    
+    public func addBreadcrumb(_ message: String,
```

Comment:
I suspect this is not thread safe. It seems like two threads could go in there at the same time. Not sure how this works in Swift, have to admit.

Reply:
Add breadcrumb feature exposed through BacktraceClient.shared and BacktraceClient is singleton class. So its thread safe and work good.

Reply:
Singleton or not, multiple threads can access a singleton at the same time, eg, it can be invoked from a foreground and background thread at roughly the same time. Can we include a unit test that verifies this?

https://medium.com/swiftcairo/avoiding-race-conditions-in-swift-9ccef0ec0b26

Reply:
Integrate Cassette for better performance and muti thread support.